ios/viewmodels/view-model-reuse.md

```diff
@@ -0,0 +1,167 @@
+
+![ProfileViewController](profile.png)
+![ReviewViewController](review.png)
+
+Let's call these two views ProfileViewController and ReviewViewController respectively.
+
+The table view cell pictured here is the same in both cases, and we would ideally like to reuse the table view cell and it's xib. In both cases, we are trying to abstract view logic out of the table view cell. However, the models that drive these two cells are very different. Let's explore two approaches of view models that can be used for this view.
+
+### Concrete view model class
+
+This simpler of the two approaches just has a concrete class for the view, which is how we would do it if the cell only had to display one kind of model.
+
+```swift
+
+struct TitleMessageCellViewModel {
+  let title: String
+  let message: String
+  let showsEditButton = false
+}
+
+class TitleMessageCell: UITableViewCell {
+  @IBOutlet private weak var titleLabel: UILabel!
+  @IBOutlet private weak var messageLabel: UILabel!
+
+  private func configure(viewModel: TitleMessageCellViewModel) {
+    titleLabel.text = viewModel.title.uppercaseString
+    messageLabel.text = viewModel.message
+    editButton.hidden = !viewModel.showsEditButton
+  }
+}
+
+```
+
+An implementation for ProfileViewController might look like this:
+
+```swift
+
+// ProfileViewController.swift
+
+func viewModel(forIndex index: Int) {
+  switch index {
+    case 0:
+      return TitleMessageCellViewModel(title: "First Name", message: profile.firstName)
+    case 1:
+      return TitleMessageCellViewModel(title: "Surname", message: profile.lastName)
+    ...
+  }
+}
+
+```
+
+This implementation abstracts out the view logic details out of the table view cell, albeit moving it to the view controller. When the view logic is trivial, this is not too bad.
+
+The implementation for ReviewViewController, however, might look like this:
+
+```swift
+
+// ReviewViewController.swift
+
+func viewModel(forIndex index: Int) {
+  switch index {
+    case 0:
+      let formatter = PhoneNumberFormatter(forCountryCode: currentCountryCode)
+      let number = formatter.formatNumber(profile.phoneNumber) ?? "-"
+      return TitleMessageCellViewModel(title: "Contact number", message: number, showsEditButton: profile.canEdit)
+    case 1:
+      let rows = topics.filter({ $0.selected }).map({ $0.title }).joinWithSeparator("\n")
+      return TitleMessageCellViewModel(title: "Visit Details", message: topics, showsEditButton: review.canEdit)
+    ...
+  }
+}
+
+```
+
+The problem with this approach is that when the view logic is trivial (ProfileViewController), we can get away with implementing view logic within the view controller, but as soon as it gets complicated(ReviewViewController), the view controller starts knowing a lot about view logic details, and we are back to view controller and view bloat.
```

Comment:
`Which VM to use is not actually dependent on anything view-related. ` <- we need to discuss this over a call. I believe that the current thinking is that it is.

As to this comment `What you actually solve is the general issue of bloated classes (classes doing too many things), not bloated VC/Vs in particular.`  <- are you saying that this would be better demonstrated by moving the whole example into a parent view model?